NPS.ID.PublicApi.DotNet.Client/Connection/WebSocketConnector.cs

```diff
@@ -0,0 +1,353 @@
+using System.Buffers;
+using System.IdentityModel.Tokens.Jwt;
+using System.Net.WebSockets;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Nordpool.ID.PublicApi.v1.Command;
+using NPS.ID.PublicApi.DotNet.Client.Connection.Extensions;
+using NPS.ID.PublicApi.DotNet.Client.Connection.Events;
+using NPS.ID.PublicApi.DotNet.Client.Connection.Messages;
+using NPS.ID.PublicApi.DotNet.Client.Connection.Options;
+using NPS.ID.PublicApi.DotNet.Client.Security;
+using NPS.ID.PublicApi.DotNet.Client.Security.Options;
+
+namespace NPS.ID.PublicApi.DotNet.Client.Connection;
+
+public class WebSocketConnector : IAsyncDisposable
+{
+    private const string SecureWebSocketProtocol = "wss";
+    private const string UnsecureWebSocketProtocol = "ws";
+
+    private bool _isConnected;
+    public bool IsConnected => _isConnected && _webSocket.State == WebSocketState.Open;
+
+    public delegate void ConnectionEstablishedEventHandler(object sender, EventArgs e);
+
+    public delegate void ConnectionClosedEventHandler(object sender, EventArgs e);
+
+    public delegate Task MessageReceivedEventHandler(object sender, MessageReceivedEventArgs e,
+        CancellationToken cancellationToken);
+
+    public delegate void ErrorEventHandler(object sender, ErrorEventArgs e);
+
+    public string ConnectionUri { get; private set; }
+    public event ConnectionEstablishedEventHandler StompConnectionEstablished;
```

Comment:
I would maybe replace events with simple listeners/handles which we can accept in ctor - the reason is that events are rarely used in modern .net and can freak ppl out :)

Reply:
I mean, we have the same implementation in ID3 Stomp Client so I would prefer to stay consistent. That's why i decided to implement it the same way.

Reply:
@PablitoCBR @paveldayneko guys what do you think about this change?

Reply:
Since we are not creating a library and just a sample which does not have to be very performant and rather should be mainly readable I am quite fine with keeping events here

Reply:
I'm voting for removing events from here as well, they feel outdated indeed. Even though it's a sample we still should put some effort in it, dropping few best approaches here and there as we've done with channels for example.

Reply:
Replaced events with callbacks passed by constructor into WebSocketConnector